compiler/ilgen/IlBuilder.cpp

```diff
@@ -1078,6 +1090,110 @@ IlBuilder::Add(TR::IlValue *left, TR::IlValue *right)
    return returnValue;
    }
 
+/*
+ * blockThrowsException:
+ * ....
+ * goto blockAfterExceptionHandler 
+ * Handler:
+ * ....
+ * goto blockAfterExceptionHandler 
+ * blockAfterExceptionHandler:
+ */
+void
+IlBuilder::appendExceptionHandler(TR::Block *blockThrowsException, TR::IlBuilder **handler, uint32_t catchType)
+   {
+   //split block after overflowCHK, and add an goto to blockAfterExceptionHandler
+   appendBlock();
+   TR::Block *blockWithGoto = _currentBlock;
+   TR::Node *gotoNode = TR::Node::create(NULL, TR::Goto);
+   genTreeTop(gotoNode);
+   _currentBlock = NULL;
+   _currentBlockNumber = -1;
+ 
+   //append handler, add exception edge and merge edge
+   *handler = createBuilderIfNeeded(*handler);
+   TR_ASSERT(*handler != NULL, "exception handler cannot be NULL\n");
+   (*handler)->_isHandler = true;
+   cfg()->addExceptionEdge(blockThrowsException, (*handler)->getEntry());
+   appendExceptionBuilder(*handler);
```

Comment:
I think this call can just be turned into AppendBuilder(*handler) without having to modify AppendBuilder at all.